cumulusci/tasks/tests/test_package_2gp.py

```diff
@@ -133,8 +106,10 @@ def task(project_config, devhub_config, org_config):
         ),
         org_config,
     )
-    task._init_devhub = mock.Mock(return_value=devhub_config)
-    task._init_task()
+    with mock.patch(
+        "cumulusci.tasks.package_2gp.get_devhub_config", return_value=devhub_config
+    ):
```

Comment:
Why don't you make the devhub_config fixture take care of doing this mocking so that you don't have to repeat it for each test?

Reply:
If the path that needs mocking may change depending on which test is running, is there a way to patch this in the `devhub_config` fixture, such that it doesn't need a parameter?

For example, in `test_package_2gp.py`, "cumulusci.tasks.package_2gp.get_devhub_config" is patched, but in `test_promote_package_version.py`, "cumulusci.tasks.salesforce.promote_package_verions.get_devhub_config" is patched.

Reply:
Oh I see, that does make it more complicated and I'm not sure I have a better suggestion for how to simplify it then.